config/config.go

```diff
@@ -219,6 +219,27 @@ type Config struct {
 	original string
 }
 
+// Secret special type for storing secrets
+type Secret string
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface for Secrets.
+func (s *Secret) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	type plain Secret
+	err := unmarshal((*plain)(s))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// MarshalYAML implements the yaml.Marshaler interface for Secrets.
+func (s Secret) MarshalYAML() (interface{}, error) {
+	if s != "" {
+		return "XXXXX", nil
```

Comment:
I wonder whether we should use a placeholder which makes it more clear that we changed the value, like `<hidden>` or so.